submissions/AnnaStoyano/Friends-app/index.js

```diff
@@ -0,0 +1,163 @@
+        this.resultsWrapper.innerHTML = '';
+        this.dipslayCards.forEach(({card}) => this.resultsWrapper.insertAdjacentElement('afterbegin', card));
+    }
+
+    filterByName() {
+        this.filterByNameAlphabet();
+        this.filterByNameNonAlphabet();
+    }
+
+    filterByNameAlphabet() {
+        const alphabetBtn = document.querySelector('.name-alphabetically');
+        alphabetBtn.addEventListener('click', () => {
+            this.dipslayCards = this.dipslayCards.sort((a, b) => a.name > b.name ? -1 : 1);
+            this.render();
+        });
+    }
+
+    filterByNameNonAlphabet() {
+        const nonAlphabetBtn = document.querySelector('.name-nonalphabetically');
+        nonAlphabetBtn.addEventListener('click', () => {
+            this.dipslayCards = this.dipslayCards.sort((a, b) => a.name > b.name ? 1 : -1);
+            this.render();
+        });
+    }
+
+    filterByAge() {
+        const ageBtn = document.querySelector('.age_btn');
+        ageBtn.addEventListener('click', () => {
+            if (!ageBtn.classList.contains('LowToHigh')) { // if button is clicked show from low to high
+                this.dipslayCards = this.dipslayCards.sort((a, b) => a.age > b.age ? -1 : 1);
+                ageBtn.classList.add('LowToHigh'); // adding class helps to find out clicking on button
+            } else {
+                this.dipslayCards = this.dipslayCards.sort((a, b) => a.age > b.age ? 1 : -1); // if button isn't clicked show from high to low
+                ageBtn.classList.remove('LowToHigh');
+            }
+            this.render();
+        });
+    }
+
+    filterByGender() {
+        const manRadio = document.querySelector('.radio-input.man');
+        const womanRadio = document.querySelector('.radio-input.woman');
+        const allRadio = document.querySelector('.radio-input.all');
+        const genderWrapper = document.querySelector('.menu_item-gender');
+        genderWrapper.addEventListener('click', () => {
+            if (womanRadio.checked) {
+                this.dipslayCards = this.dipslayCards.filter(card => card.user.gender == 'female');
+            }
+            if (manRadio.checked) {
+                this.dipslayCards = this.dipslayCards.filter(card => card.user.gender == 'male');
+            }
+            if (allRadio.checked) {
+                this.dipslayCards = this.cards;
+            }
+            this.render();
+        });
+    }
+
+    filterByCity() {
+        const cityInput = document.querySelector('.text-input--city');
+        const cityWrapper = document.querySelector('.menu_item-city');
+        cityWrapper.addEventListener('click', () => {
+            this.previouseDisplay = this.dipslayCards;
+        });
+        cityInput.addEventListener('input', ({target}) => {
+            this.dipslayCards = this.dipslayCards.filter(card => card.user.location.city.startsWith(target.value));
+            if (this.dipslayCards.length == 0 || target.value == '') {
```

Comment:
So if nothing was found - show all results? 🤔 Not the best choice. You should show empty results.

Also, if it's an empty request - why do you filter? 🤔